src/github.com/getlantern/idletiming/idletiming_conn.go

```diff
@@ -202,30 +247,50 @@ func (c *IdleTimingConn) SetDeadline(t time.Time) error {
 }
 
 func (c *IdleTimingConn) SetReadDeadline(t time.Time) error {
-	// zero value means no time out
-	if t.IsZero() {
-		t = epoch
+	c.closeMutex.RLock()
+	defer c.closeMutex.RUnlock()
+
+	if err := c.checkClosed(nil); err != nil {
+		return err
 	}
+
 	atomic.StoreInt64(&c.readDeadline, t.UnixNano())
 	return nil
 }
 
 func (c *IdleTimingConn) SetWriteDeadline(t time.Time) error {
-	// zero value means no time out
-	if t.IsZero() {
-		t = epoch
+	c.closeMutex.RLock()
+	defer c.closeMutex.RUnlock()
+
+	if err := c.checkClosed(nil); err != nil {
+		return err
 	}
+
 	atomic.StoreInt64(&c.writeDeadline, t.UnixNano())
 	return nil
 }
 
 func (c *IdleTimingConn) markActive(n int) bool {
 	if n > 0 {
-		c.activeCh <- true
+		select {
```

Comment:
@myleshorton I just noticed this, so snuck in a fix.  We don't want to risk markActive blocking if the run loop isn't reading from the activeCh because it's blocking on something else (like Close()).